### PR TITLE
tooltip with html support and appendToBody

### PR DESCRIPTION
**Pull request by erashu212**
Hi @valorkin ,

Please review and merge this PR.

Thanks

### CODE REVIEW

**Comment by erashu212** [comment]
Hi @valorkin 

Any update on this?

**Comment by likerRr** [comment]
@valorkin @erashu212 I'm not sure, but does this PR really implement `appendToBody` functionality? As I can see in [source code](https://github.com/valor-software/ng2-bootstrap/blob/48234e38f8ddcd69561cb1c0c02814566b5dc9cb/components/position.ts) now `appendToBody` just changes the way of how the position of tooltip will be calculated. But expected behavior with `appendToBody=true` is that tooltip will be attached to the `body` element instead of tooltips's parent

**Comment by erashu212** [comment]
Hi @likerRr ,

Is not that working?

**Comment by likerRr** [comment]
Hi, @erashu212. I haven't checked it yet, but `appendToBody` should be able to solve problem when a tooltip is placed inside overflowed container.

**Comment by erashu212** [comment]
Hi @likerRr ,

Please try and if possible create a plunker so we can have look into this.

Thanks

**Comment by likerRr** [comment]
@erashu212 I haven't ability to try your PR now. But [here is the plunker](http://plnkr.co/edit/k0Dv2Wl1H5bQxunbGPr3?p=preview) with the common issue with the `tooltip` (it cuts inside an overflowed container) and this should be solved with `appendToBody` enabled. Could you check if your PR fixes such situation?

**Comment by erashu212** [comment]
hi @likerRr ,

Thank you for pointing out this mistake. I have created an updated version of [plnkr](http://plnkr.co/edit/ns02nXoq9K5qizx1u121?p=preview).

Thanks,

File: components/tooltip/tooltip.directive.ts [hunk 1]
```diff
@@ -1,68 +1,81 @@
...
     ]);
 
     this.tooltip = this.loader
       .loadNextToLocation(TooltipContainerComponent, this.viewContainerRef, binding)
-      .then((componentRef:ComponentRef<any>) => {
-        return componentRef;
-      });
+      .then((componentRef: ComponentRef<any>) => {
+      if (this.appendToBody) {
+        let dialogElement = componentRef.location.nativeElement;
```
Comment (likerRr) [inline components/tooltip/tooltip.directive.ts]: > `let dialogElement = componentRef.location.nativeElement;`

Why the name of tooltip element is `dialogElement`? Unsuccessful copy-paste? :)

**Comment by likerRr** [comment]
@erashu212 Nice work, now it looks much better :)

File: components/tooltip/tooltip.directive.ts (hunk 1 above)
Comment (erashu212) [inline components/tooltip/tooltip.directive.ts]: Hi @likerRr , 

You are right unsuccessful copy-paste :)

**Comment by likerRr** [comment]
@valorkin any updates? html content and body as container are very requested features

**Comment by valorkin** [comment]
append to body is most anticipated feature with ng2
because there no right approach to it
check modals for example

I will have to tripple check how this PR works, before merging

**Comment by erashu212** [comment]
@valorkin : 

Did you manage to look into this?

**Comment by adrianfaciu** [comment]
Maybe it would be an idea to split this in two ? One part with html content and the other with append to body. In this way html content could be merged, and append to body could be still looked into.

**Comment by likerRr** [comment]
`BrowserDomAdapter` is [no longer exported as public API](https://github.com/angular/angular/issues/8509) (it's used in this PR)

**Comment by adrianfaciu** [comment]
If needed I can create a pull request with the html content part only, I'm already using this from my fork. From my point of view the sooner we add more features the better :smile:

**Comment by Dinistro** [comment]
@adrianfaciu Would be great

**Comment by valorkin** [comment]
Ok, some things now much easier to achieve in angular2
like get document.body in a nice way (see components helper get document)
other things is using private stuff ( like browser dom adapter) will block 
system.js users from using ng2-bootstrap
- will forbid server side rendering

Thanks you for you help and time! :)
Sorry that I can not merge it

please join slack: https://www.hamsterpad.com/chat/ng2

**Pull request closed without merging**